src/utilities/vc_ins_utilities.cpp

```diff
@@ -0,0 +1,878 @@
+
+            for (Box<NDIM>::Iterator it(patch_box); it; it++)
+            {
+                CellIndex<NDIM> ci(it());
+                const double phi = (*ls_data)(ci);
+
+                double h_phi = IBTK::smooth_heaviside(phi, alpha);
+                (*mu_data)(ci) = d_mu_gas + (d_mu_liquid - d_mu_gas) * h_phi;
+            }
+        }
+    }
+
+    return;
+} // setViscosityPatchData2PhaseFlows
+
+void
+SetFluidProperties::setViscosityPatchData3PhaseFlows(int mu_idx,
+                                                     Pointer<Variable<NDIM> > mu_var,
+                                                     SAMRAI::tbox::Pointer<HierarchyMathOps> hier_math_ops,
+                                                     const int /*cycle_num*/,
+                                                     const double time,
+                                                     const double current_time,
+                                                     const double new_time)
+{
+    // Get the current level set information
+    VariableDatabase<NDIM>* var_db = VariableDatabase<NDIM>::getDatabase();
+    int ls_solid_idx = -1;
+    int ls_gas_idx = -1;
+
+    if (IBTK::rel_equal_eps(time, current_time))
+    {
+        ls_solid_idx = var_db->mapVariableAndContextToIndex(d_ls_solid_var, d_adv_diff_solver->getCurrentContext());
+    }
+    else if (IBTK::rel_equal_eps(time, new_time))
+    {
+        ls_solid_idx = var_db->mapVariableAndContextToIndex(d_ls_solid_var, d_adv_diff_solver->getNewContext());
+    }
+    else
+    {
+        TBOX_ERROR("This statement should not be reached");
+    }
+
+    if (IBTK::rel_equal_eps(time, current_time))
+    {
+        ls_gas_idx = var_db->mapVariableAndContextToIndex(d_ls_gas_var, d_adv_diff_solver->getCurrentContext());
+    }
+    else if (IBTK::rel_equal_eps(time, new_time))
+    {
+        ls_gas_idx = var_db->mapVariableAndContextToIndex(d_ls_gas_var, d_adv_diff_solver->getNewContext());
+    }
+    else
+    {
+        TBOX_ERROR("This statement should not be reached");
+    }
+
+    Pointer<PatchHierarchy<NDIM> > patch_hierarchy = hier_math_ops->getPatchHierarchy();
+    const int coarsest_ln = 0;
+    const int finest_ln = patch_hierarchy->getFinestLevelNumber();
+
+    Pointer<CellVariable<NDIM, double> > mu_cc_var = mu_var;
+    if (mu_cc_var)
+    {
+        for (int ln = coarsest_ln; ln <= finest_ln; ++ln)
+        {
+            Pointer<PatchLevel<NDIM> > level = patch_hierarchy->getPatchLevel(ln);
+            for (PatchLevel<NDIM>::Iterator p(level); p; p++)
+            {
+                Pointer<Patch<NDIM> > patch = level->getPatch(p());
+                Pointer<CartesianPatchGeometry<NDIM> > patch_geom = patch->getPatchGeometry();
+                const double* const patch_dx = patch_geom->getDx();
+                const double alpha = d_num_solid_interface_cells * patch_dx[0];
+                const double beta = d_num_gas_interface_cells * patch_dx[1];
```

Comment:
This looks suspicious to me. Why is one set with `dx` and one with `dy`?

Reply:
Thank you. I have now modified the code to use the correct implementation.